### PR TITLE
Generate analysis file scripts added

### PR DESCRIPTION
**Pull request by runjin326**
#### What scientific question is your analysis addressing?
Pull in the generate files script from OpenPBTA 

#### What was your approach?
1) Grab the generate analysis files bash script from OpenPBTA
2) Modify the one script to two scripts - one for toolkit and one for repo

The one for toolkit will run the following modules using the merged SNV, CNV, expression, and fusion files.
```
fusion-filtering
GISTIC
focal CN preparation
```
The one for repo will run the following modules after pulling in files from the toolkit run as well
```
independent-samples
tmb-calculation
fusion-summary
```

#### What GitHub issue does your pull request address?
NA

#### Which areas should receive a particularly close look?
Note that for independent samples module, unlike OpenPBTA which uses base histology file, for OpenPedCan, I set it to use full histologies file (to allow cohort etc)

#### Is there anything that you want to discuss further?
1) Is it ok to use full histology for independent samples 
2) I have not run the script - do we want to run the repo one to try it out?
3) How can we QC the changes? 

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?
Yes.

#### What types of results are included (e.g., table, figure)?
NA

#### What is your summary of the results?
NA

#### Reproducibility Checklist

- [ ] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [ ] This analysis has been added to continuous integration.

#### Documentation Checklist

- [ ] This analysis module has a `README` and it is up to date.
- [ ] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [x] The analytical code is documented and contains comments.

### CODE REVIEW

File: scripts/generate-analysis-files-toolkit.sh [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  cp ${analyses_dir}/run-gistic/results/cnv-consensus-gistic-only.seg.gz ${compiled_dir}
+  
+  # Run step that generates "most focal CN" files (annotation) using the *BASE* histology file
+  echo "Run focal CN file preparation"
+  OPENPBTA_BASE_RELEASE=1 bash ${analyses_dir}/focal-cn-file-preparation/run-prepare-cn.sh
+  
+  # Copy over focal CN
+  cp ${analyses_dir}/focal-cn-file-preparation/results/consensus_wgs_plus_cnvkit_wxs_autosomes.tsv.gz ${compiled_dir}
+  cp ${analyses_dir}/focal-cn-file-preparation/results/consensus_wgs_plus_cnvkit_wxs_x_and_y.tsv.gz ${compiled_dir}
+  cp ${analyses_dir}/focal-cn-file-preparation/results/consensus_wgs_plus_cnvkit_wxs.tsv.gz ${compiled_dir}
```
Comment (jharenza) [inline scripts/generate-analysis-files-toolkit.sh:62]: Do we need to copy over all 3 or can we just use the combined file? Did we update analysis modules to use combined or are we still using separate files for some things?

File: scripts/generate-analysis-files-toolkit.sh [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  # Run GISTIC step -- only the part that generates ZIP file
+  echo "Run GISTIC"
+  # Run a step that subs ploidy for NA to allow GISTIC to run
+  Rscript ${analyses_dir}/run-gistic/scripts/prepare_seg_for_gistic.R \
+  --in_consensus ${data_dir}/cnv-consensus.seg.gz \
+  --out_consensus ${analyses_dir}/run-gistic/results/cnv-consensus-gistic-only.seg.gz \
+  --histology ${data_dir}/histologies-base.tsv
+  
+  # This will use the file that just got generated above
+  bash ${analyses_dir}/run-gistic/scripts/run-gistic-opentargets.sh
```
Comment (jharenza) [inline scripts/generate-analysis-files-toolkit.sh:49]: Is this another situation in which we created two run scripts in a module? If so, we should remove one to keep it generalized. The history will be in the commits anyway if we wanted to go back.

File: scripts/generate-analysis-files-repo.sh [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+# Copy over fusion summary
+cp ${analyses_dir}/fusion-summary/results/* ${compiled_dir}
+
+# Run TMB step
+echo "Create TMB results"
+bash ${analyses_dir}/tmb-calculation/run_tmb_calculation.sh
+
+# Copy over TMB results
+cp ${analyses_dir}/tmb-calculation/results/snv-mutation-tmb-coding.tsv ${compiled_dir}
+cp ${analyses_dir}/tmb-calculation/results/snv-mutation-tmb-all.tsv ${compiled_dir}
```
Comment (jharenza) [inline scripts/generate-analysis-files-repo.sh:48]: Do we need to add TCGA tmb as well or is that within this file?

**Review by jharenza — COMMENTED** [review]
Few questions - see comments

File: scripts/generate-analysis-files-toolkit.sh (hunk 1 above)
Comment (runjin326) [reply scripts/generate-analysis-files-toolkit.sh:62]: This module where we generate seg-like files for PedCBio needs all 3 files: https://github.com/PediatricOpenTargets/OpenPedCan-analysis/tree/dev/analyses/pedcbio-cnv-prepare

File: scripts/generate-analysis-files-toolkit.sh (hunk 2 above)
Comment (runjin326) [reply scripts/generate-analysis-files-toolkit.sh:49]: This was coded like this in OpenPBTA script - the module itself has one bash script in the base of the module and another script in the `scripts` folder. They are also organized like that in OpenPBTA.

File: scripts/generate-analysis-files-repo.sh (hunk 3 above)
Comment (runjin326) [reply scripts/generate-analysis-files-repo.sh:48]: This module currently does not calculate TCGA - do we need a module update to include TCGA?

Comment (jharenza) [reply scripts/generate-analysis-files-repo.sh:48]: Oh, right. We don't have TCGA DNA in our releases yet. Will hold on that until needed.

File: scripts/generate-analysis-files-toolkit.sh (hunk 2 above)
Comment (jharenza) [reply scripts/generate-analysis-files-toolkit.sh:49]: Hmm, ok. Can leave for now

**Review by jharenza — APPROVED** [review]

**Pull request merged**